lib/core/src/lib/icon/icon.component.ts

```diff
@@ -15,22 +15,26 @@
  * limitations under the License.
  */
 
-import { Component, Input, ViewEncapsulation, ChangeDetectionStrategy } from '@angular/core';
+import { Component, Input, ViewEncapsulation, ChangeDetectionStrategy, inject } from '@angular/core';
 import { ThemePalette } from '@angular/material/core';
 import { MatIconModule } from '@angular/material/icon';
-import { NgIf } from '@angular/common';
+import { ICON_ALIAS_MAP_TOKEN } from './icon-alias-map.token';
+
+export const DEFAULT_ICON_VALUE = 'settings';
 
 @Component({
     selector: 'adf-icon',
-    imports: [MatIconModule, NgIf],
+    imports: [MatIconModule],
     templateUrl: './icon.component.html',
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush,
     host: { class: 'adf-icon' }
 })
 export class IconComponent {
-    private _value = '';
-    private _isCustom = false;
+    private readonly ALIAS_MAP = inject(ICON_ALIAS_MAP_TOKEN, { optional: true });
```

Comment:
this is not injecting multiple entries I think... the docs say you can provide multiple times but this inject will take only 1